src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1073,7 +1073,7 @@ private void DoGraphAutoLayout(object o)
                 // Do not process nodes within groups
                 if (group == null)
                 {
-                    graph.AddNode(x.GUID, x.Width, x.Height, x.Y);
+                    graph.AddNode(x.GUID, x.Width, x.Height, x.Y, x.InPorts.Count);
```

Comment:
Minor point but `x` here is a bit confusing with x-coordinate. Could you rename it to `n` for instance to denote a node?